source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -204,22 +205,10 @@ def _getCurrentOffsetInThisLine(self, lineInfo):
 		find word boundaries, but UIA only allows for relative movement.
 		"""
 		charInfo = self.copy()
-		res = 0
-		chars = None
-		while charInfo.compareEndPoints(
-			lineInfo,
-			"startToEnd"
-		) <= 0:
-			charInfo.expand(textInfos.UNIT_CHARACTER)
-			chars = charInfo.move(textInfos.UNIT_CHARACTER, -1) * -1
-			if chars != 0 and charInfo.compareEndPoints(
-				lineInfo,
-				"startToStart"
-			) >= 0:
-				res += chars
-			else:
-				break
-		return res
+		charInfo.setEndPoint(lineInfo, "startToStart")
+		text = charInfo.text
+		offset = textUtils.WideStringOffsetConverter(text).wideStringLength - 1
```

Comment:
At first glance, it's not entirely clear to me where the -1 is coming from here.